pyispyb/core/routes/samples.py

```diff
@@ -147,3 +171,46 @@ def get_sample(
         return samples.first
     except IndexError:
         raise HTTPException(status_code=404, detail="Sample not found")
+
+
+@router.patch(
+    "/{blSampleId}",
+    response_model=schema.Sample,
+    responses={404: {"description": "No such sample"}},
+)
+def update_sample(
+    sample: schema.SampleUpdate,
+    blSampleId: int = Depends(filters.blSampleId),
+) -> models.BLSample:
+    """update a sample"""
+    try:
+        sample = crud.update_sample(blSampleId=blSampleId, sample=sample)
+        return sample
+    except IndexError:
+        raise HTTPException(status_code=404, detail="Sample not found")
```

Comment:
You might like a final `except Exception` to catch anything else that might go wrong as is done on the shipment route:

```
    except Exception:
        logger.exception(
            f"Could not update shipping `{shippingId}` with payload `{shipping}`"
        )
        raise HTTPException(status_code=400, detail="Could not update shipment")
```